app/src/main/java/com/xmartlabs/template/helper/DateHelper.java

```diff
@@ -43,307 +44,141 @@
-    } catch (ParseException e) {
-      throw new RuntimeException(e);
-    }
+  private static DateTimeFormatter getDeviceLongDateFormat() {
+    SimpleDateFormat timeFormat = (SimpleDateFormat) DateFormat.getLongDateFormat(BaseProjectApplication.getContext());
+    return DateTimeFormatter.ofPattern(timeFormat.toPattern());
   }
 
   /**
-   * Converts a Date object to a string representation
+   * Converts a TemporalAccessor object to a string representation
    *
-   * @param date Date object
-   * @param df   Date format
+   * @param temporalAccessor TemporalAccessor object
+   * @param df               Date format
    * @return Date string formatted on the right way
    */
   @NonNull
-  public static String dateToString(@NonNull Date date, @NonNull DateFormat df) {
-    return df.format(date);
+  public static String dateToString(@NonNull TemporalAccessor temporalAccessor, @NonNull DateTimeFormatter df) {
+    return df.format(temporalAccessor);
   }
 
   /**
-   * Converts a string representation of a date to its Date object.
+   * Converts a string representation of a date to its LocalDateTime object.
    *
    * @param dateAsString Date in string format
    * @param df           Date format
    * @return Date
    */
   @Nullable
-  public static Date stringToDate(@NonNull String dateAsString, @NonNull DateFormat df) {
+  public static LocalDateTime stringToLocalDateTime(@NonNull String dateAsString, @NonNull DateTimeFormatter df) {
     try {
-      df.setTimeZone(UTC);
-      return df.parse(dateAsString);
-    } catch (ParseException e) {
+      return LocalDateTime.parse(dateAsString, df);
+    } catch (Exception e) {
       return null;
     }
   }
 
   /**
    * Sets the <code>newDate</code> year, month and day of month onto <code>oldDateAndTime</code>
+   *
    * @param oldDateAndTime the date to update
-   * @param newDate the new date that will be used to update <code>oldDateAndTime</code>
-   * @return a new <code>Date</code> instance with the time parameters of the <code>oldDateAndTime</code> and the
+   * @param newDate        the new date that will be used to update <code>oldDateAndTime</code>
+   * @return a new <code>LocalDateTime</code> instance with the time parameters of the <code>oldDateAndTime</code> and
+   * the
    * date parameters of the <code>newDate</code>
    */
   @NonNull
-  public static Date setDateButKeepTime(@NonNull Date oldDateAndTime, @NonNull Date newDate) {
-    Calendar oldDateAndTimeCalendar = DateUtils.toCalendar(oldDateAndTime);
-
-    Calendar newDateCalendar = DateUtils.toCalendar(newDate);
-
-    oldDateAndTimeCalendar.set(
-        newDateCalendar.get(Calendar.YEAR),
-        newDateCalendar.get(Calendar.MONTH),
-        newDateCalendar.get(Calendar.DAY_OF_MONTH)
-    );
-
-    return oldDateAndTimeCalendar.getTime();
+  public static LocalDateTime setDateButKeepTime(@NonNull LocalDateTime oldDateAndTime, @NonNull LocalDateTime newDate) {
+    LocalDate date = LocalDate.from(newDate);
```

Comment:
I think this function has not much sense anymore with the new api. I'll remove it. make sense?